src/Workspaces/Core/Portable/Serialization/SerializerService_Reference.cs

```diff
@@ -105,17 +122,21 @@ protected virtual MetadataReference ReadMetadataReferenceFrom(ObjectReader reade
         throw ExceptionUtilities.UnexpectedValue(type);
     }
 
-    protected virtual void WriteAnalyzerReferenceTo(AnalyzerReference reference, ObjectWriter writer)
+    protected virtual void WriteAnalyzerReferenceTo(
+        AnalyzerReference reference,
+        ObjectWriter writer,
+        bool forTesting)
```

Comment:
Rather than a "forTesting" flag, just do the validation in DEBUG? Because I think it's still a valid sanity check there, right?